airflow/providers/opentelemetry/plugins/otel_listener.py

```diff
@@ -0,0 +1,90 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import os
+
+# import os
+from typing import TYPE_CHECKING
+
+from airflow.configuration import conf
+from airflow.listeners import hookimpl
+from airflow.providers.opentelemetry.hooks.otel import OtelHook
+
+if TYPE_CHECKING:
+    from airflow.datasets import Dataset
+    from airflow.models.dagrun import DagRun
+    from airflow.models.taskinstance import TaskInstance
+    from airflow.utils.state import TaskInstanceState
+
+"""
+from opentelemetry import trace
+from opentelemetry.sdk.trace import Span, Tracer
+from opentelemetry.trace import NonRecordingSpan, TraceFlags
+from airflow.configuration import conf
+from airflow.providers.otel.hooks.otel import OtelHook
+"""
+
+_OTEL_CONN = conf.get("otel", "conn_id", fallback=os.getenv("OTEL_CONN_ID", OtelHook.default_conn_name))
+
+
+@hookimpl
+def on_task_instance_running(previous_state: TaskInstanceState, task_instance: TaskInstance, session):
+    """Trigger task state changes to RUNNING."""
```

Comment:
Hmm.- are those going to be implemented? Why the listener is here?

Reply:
Yes, these are to be implemented. The purpose of these listeners are actually quite interesting. There are versions of Airflow that does not have OTEL trace capabilities (due to their low versions), so one of the purpose of providing these 'optional' listener is that user can send a very similar trace that OTEL airflow tracing can do using these listeners.

So, users would have the option to turn on the listeners and use that in that case. Listeners would use Otel Hook to configure the connectivity in this case.